### PR TITLE
Add method of ignoring classes, constants and functions from global namespace

### PR DESCRIPTION
**Pull request by padraic**
Reset of original PR. On a parallel to the PHP 7 fix using a custom attribute set on nodes to ignore PHP/SPL/Extension classes. Same is also needed for:

- [x] Apply to FQN constants, e.g. \PHP_EOL vs PHP_EOL.
- [x] Apply to FQN functions, should be even rarer then constants but possible, e.g. \str_replace().
- [x] Test to non-FQNs to rule out impacts there.
- [x] Test for return type declarations for consistency.

Original used name collection, this instead switches to a simple convention of skipping single part FQNs or non-FQNs. e.g. `\Something` or `Something` vs `\MyNS\Something` or `MyNS\Something`. For now that suggests a need to allow configuration of exceptions for userland code which is part of the global namespace (presumably legacy code/packages).

### CODE REVIEW

File: .styleci.yml [hunk 1]
```diff
@@ -10,4 +10,4 @@ disabled:
 
 finder:
   exclude:
-    - 'fixtures'
+    - 'reserved_classes.php'
```
Comment (padraic) [inline .styleci.yml]: Exclude not working at present, will need to check or just integrate into the test as a string.

File: tests/Fixtures/reserved_classes.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (theofidry) [inline tests/Fixtures/reserved_classes.php]: I think you could move that under `ScoperTest:: provideValidFiles()`

File: src/NodeVisitor/FullyQualifiedNamespaceUseScoperNodeVisitor.php [hunk 3]
```diff
@@ -37,6 +37,11 @@ public function __construct($prefix)
     public function enterNode(Node $node)
     {
         if ($node instanceof FullyQualified) {
```
Comment (theofidry) [inline src/NodeVisitor/FullyQualifiedNamespaceUseScoperNodeVisitor.php]:
```php
if (
    $node instanceof FullyQualified
    (false === $node->hasAttribute('phpscoper_ignore') || false === $node->getAttribute('phpscoper_ignore'))
) {
    return new Name(Name::concat($this->prefix, (string) $node));
}
```

File: src/NodeVisitor/UseNamespaceScoperNodeVisitor.php [hunk 4]
```diff
@@ -38,9 +38,15 @@ public function __construct($prefix)
     public function enterNode(Node $node)
     {
         if ($node instanceof UseUse
-            && $node->hasAttribute('parent') && false === ($node->getAttribute('parent') instanceof  GroupUse)
+            && $node->hasAttribute('parent')
+            && false === ($node->getAttribute('parent') instanceof GroupUse)
             && $this->prefix !== $node->name->getFirst()
         ) {
+            if ($node->hasAttribute('phpscoper_ignore')
```
Comment (theofidry) [inline src/NodeVisitor/UseNamespaceScoperNodeVisitor.php]: like above I think this can be moved to a single if statement

**Comment by theofidry** [comment]
Some small CS stuff nitpicking, all good otherwise

**Pull request merged**